File: s1/c3.cpp
#include <cstdio>
#include <string>
#include "xor_utils.h"

using namespace std;

void run(const string& t1) {
  auto ret = bruteXor_eng_char(unhexlify(t1)).first;
  for(auto i = ret.begin(); i != ret.end(); i++) {
    printf("%s\n", i->decrypted.c_str());
  }
}

int main() {
	string t1="1b37373331363f78151b7f2b783431333d78397828372d363c78373e783a393b3736";
	run(t1);
	return 0;
}

File: s1/c3.py
#!/usr/bin/python3

import binascii
import itertools
from py_common import languageScore
from py_common.cryptoUtils import strxor
from collections import Counter

def main():
    t1 = binascii.unhexlify(b"1b37373331363f78151b7f2b783431333d78397828372d363c78373e783a393b3736")
    max_v = 0
    ret = str()
    for x in range(0, 256):
        res = strxor(t1, chr(x).encode('latin1'))
        value = languageScore.freqScore(res)
        if value > max_v:
            ret = res
            max_v = value
    print("Max: ",max_v," ",ret)

if __name__ == '__main__':
    main()
